tensorflow/core/ops/math_ops.cc

```diff
@@ -131,90 +131,102 @@ tf.complex_abs(x) ==> [5.25594902, 6.60492229]
 
 // Declares cwise unary operations signature: 't -> 't
 #define UNARY()                      \
-  Input("x: T").Output("y: T").Attr( \
-      "T: {half, float, double, int32, complex64, int64}")
+  Input("x: T").Output("y: T")
```

Comment:
Good idea. However, there is one more kind:

```
REGISTER_OP("Lgamma")
    .UNARY()
    .Attr("T: {half, float, double}")
    .Doc(R"doc(
Computes the log of the absolute value of `Gamma(x)` element-wise.
)doc");
```
Should we have three MACROS in total?

- UNARY:  {half, float, double, int32, complex64, int64}
- UNARY_FP:  {half, float, double, int64}
- UNARY_FPC:  {half, float, double, complex64}

I think it's getting complicated. For somehow, just adding `int32, int64` in the cwise_op* might be simpler.

` REGISTER4(UnaryOp, CPU, "Cos", functor::cos, float, Eigen::half, double, int32, int64, complex64);
`
 It won't add much as we discussed, but does it hurt? I guess there is no backwards compatibility issue, and no need to add two more macros.